### ISSUE TITLE
Cursor option for "caret_background_color" fails to activate

### ISSUE THREAD

**Issue by inhalt120g** [godotengine/godot#86281]
### Tested versions

v4.2.1.stable.official [b09f793f5]

### System information

Godot v4.2.1.stable - macOS 14.2.0 - Vulkan (Forward+) - integrated Apple M1 - Apple M1 (8 Threads)

### Issue description

The setting for the caret fails to render. Mentioned here:
https://github.com/godotengine/godot-proposals/issues/8473#issuecomment-1845930636

### Steps to reproduce

In Editor Settings, find the background color setting for the caret. Change it to whatever, but it will have no effect.
![image](https://github.com/godotengine/godot/assets/106015575/477be6c3-e699-45bf-bfae-69492ade0916)

### Minimal reproduction project (MRP)

Godot IDE

**Comment by jsjtxietian** [comment]
Indeed, unlike `caret_color`, the `caret_background_color` is never used in `text_edit.cpp`

**Comment by TheSofox** [comment]
In the linked issue @Paulb23 said:
> `caret_background_color` should be the colour of the text behind the caret when block caret is enabled, but currently there's a bug where it's not actually used anywhere.

I assume "block caret" is when you hit the `Ins` key on your keyboard and the caret switches to something that underlines a letter at a time. 

Can I confirm that this setting is meant to change the colour of the text character that the cursor and not the background of the selected letter?

To put it another way, Option 1 or 2?

![image](https://github.com/user-attachments/assets/8f2f36f5-8b74-42f6-a291-f2ba6148e368)

**Comment by inhalt120g** [comment]
> I assume "block caret" is when you hit the Ins key on your keyboard and the caret switches to something that underlines a letter at a time.

I was not able to test this since Macs don't use Insert key (there are ways to add it but it's not a 1:1 mapping as on Windows. Related discussion [link](https://apple.stackexchange.com/questions/425957/where-is-the-insert-button-on-the-apple-keyboard)). But I don't think it's relevant here.


> Can I confirm that this setting is meant to change the colour of the text character that the cursor and not the background of the selected letter?
> To put it another way, Option 1 or 2?

The name of the option is "Caret background color", so it looks like option 2.

**Comment by TheSofox** [comment]
> The name of the option is "Caret background color", so it looks like option 2.

You see, that's exactly what I was thinking, it makes sense.

But check the documentation: https://docs.godotengine.org/en/stable/classes/class_textedit.html#class-textedit-theme-color-caret-background-color

> [Color](https://docs.godotengine.org/en/stable/classes/class_color.html#class-color) of the text behind the caret when using a block caret.

This seems to imply Option 1.

Meanwhile within the editor.
![284141749-c22d414a-af72-4ed7-8cf9-6ede02e864d8](https://github.com/user-attachments/assets/ef7b4438-dec1-462b-a11e-dfc5428cdc87)

`The script editor's caret background color.` seems to imply Option 2.

I'm genuinely not sure which one it is.
